packages/assets-controllers/src/multicall.ts

```diff
@@ -0,0 +1,398 @@
+import { Contract } from '@ethersproject/contracts';
+import type { Web3Provider } from '@ethersproject/providers';
+import type { Hex } from '@metamask/utils';
+
+import { reduceInBatchesSerially } from './assetsUtil';
+
+// https://github.com/mds1/multicall/blob/main/deployments.json
+const MULTICALL_CONTRACT_BY_CHAINID = {
```

Comment:
The [multicall3](https://github.com/mds1/multicall) smart contract is introduced.  

Simple but useful utility here.  It accepts an array of calls, and executes them via a single RPC multicall if the chain supports it.  And falls back to N requests otherwise.

I was initially using [eth-balance-checker](https://github.com/wbobeirne/eth-balance-checker), but it was worse in 2 areas.

- It requires the same list of tokens to be checked across all accounts.  Which can blow up into a lot of unnecessary checks.  e.g. you're watching vitalik.eth and now you have to check 1000 tokens on every account.

- multicall3 is deployed on a _lot_ more networks (~250)